src/modules/audio/audio.c

```diff
@@ -30,21 +36,18 @@ struct Source {
   bool playing;
   bool looping;
   bool spatial;
+  float pose[16];
+  int output_channel_count;
 };
 
-typedef struct {
-  bool (*init)(void);
-  void (*destroy)(void);
-  void (*apply)(Source* source, const void* input, float* output, uint32_t frames);
-} Spatializer;
-
 static struct {
   bool initialized;
   ma_context context;
-  AudioConfig config[2];
-  ma_device devices[2];
-  ma_mutex locks[2];
+  AudioConfig config[AUDIO_TYPE_COUNT];
+  ma_device devices[AUDIO_TYPE_COUNT];
+  ma_mutex locks[AUDIO_TYPE_COUNT];
```

Comment:
Using a lockfree ringbuffer for capture is really cool, maybe we could just have a single `state.lock` now, unless we anticipate future concurrency issues with audio capture.